packages/ui/components/overlays/test/OverlayController.test.js

```diff
@@ -629,6 +654,30 @@ describe('OverlayController', () => {
         ctrl.contentNode.dispatchEvent(new KeyboardEvent('keyup', { key: 'Escape' }));
         expect(ctrl.isShown).to.be.true;
       });
+
+      it('hides all overlays when [escape] is pressed on outside element', async () => {
```

Comment:
I wonder if the approach still works if we have two ctrls: ctrl1 with hidesOnOutsideEsc and ctrl2 with hidesOnEsc. And ctrl2 is nested in ctrl1 (its contentNode is nested).

Would ctrl1 still be autonomous and be closed? (I suspect not, as `_blocksEscKeyHandler` works as a global option (on OverlaysManager level) now).

Reply:
Maybe we can solve this in a different way?
You could for instance check whether the esc event came from a child in the dom. If so, and we have `closeOnEsc` and not `closeOnOutsideEsc`, we should not close ourselves.